### ISSUE TITLE
Gem Load Error is: uninitialized constant Int

### ISSUE THREAD

**Issue by firstsano** [rom-rb/rom-rails#90]
Fresh project, ruby-2.4.0, rails-5.2.0. After installing rom-rails gem I can't start project and can't run specs. I'm getting:

```
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:84:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'rom-rails'. (Bundler::GemRequireError)
Gem Load Error is: uninitialized constant Int
Backtrace for gem load error is:
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `block in load_missing_constant'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/active_support.rb:8:in `without_bootsnap_cache'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/active_support.rb:53:in `rescue in load_missing_constant'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/active_support.rb:43:in `load_missing_constant'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-core-4.2.0/lib/rom/relation/curried.rb:38:in `<class:Curried>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-core-4.2.0/lib/rom/relation/curried.rb:19:in `<class:Relation>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-core-4.2.0/lib/rom/relation/curried.rb:10:in `<module:ROM>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-core-4.2.0/lib/rom/relation/curried.rb:9:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-core-4.2.0/lib/rom/relation.rb:16:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-core-4.2.0/lib/rom/core.rb:11:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-core-4.2.0/lib/rom-core.rb:1:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-4.2.0/lib/rom.rb:1:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/rom-rails-1.1.1/lib/rom-rails.rb:1:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:71:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:71:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:70:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:79:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:81:in `block (2 levels) in require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:76:in `each'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:76:in `block in require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:65:in `each'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:65:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler.rb:114:in `require'
/home/riky/work/warehouse/config/application.rb:18:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:145:in `block in perform'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:142:in `tap'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:142:in `perform'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/command/base.rb:65:in `perform'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/command.rb:46:in `invoke'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands.rb:18:in `<main>'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
/home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
bin/rails:4:in `<main>'
Bundler Error Backtrace:
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:80:in `block (2 levels) in require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:76:in `each'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:76:in `block in require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:65:in `each'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler/runtime.rb:65:in `require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bundler-1.16.2/lib/bundler.rb:114:in `require'
        from /home/riky/work/warehouse/config/application.rb:18:in `<main>'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:145:in `block in perform'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:142:in `tap'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:142:in `perform'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/command/base.rb:65:in `perform'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/command.rb:46:in `invoke'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/railties-5.2.0/lib/rails/commands.rb:18:in `<main>'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
        from /home/riky/.rvm/gems/ruby-2.4.0@warehouse/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
        from bin/rails:4:in `<main>'
```

**Comment by cflipse** [comment]
Can you generate a minimal repo that demonstrates this and link here?

**Comment by wmaciejak** [comment]
@firstsano Fix your version of dry-types to `0.12.2`. 

@cflipse This issue isn't related to `rom-rails`, but to rom itself. It doesn't support dry-types `0.13.0`
https://github.com/rom-rb/rom/pull/498

**Comment by firstsano** [comment]
Sorry for the delay. Yeah, it works, my bad!
Thank you for the efforts.
